LibreNMS/Util/StringHelpers.php

```diff
@@ -28,7 +28,25 @@
 class StringHelpers
 {
     /**
-     * Shorten text over 50 chars, if shortened, add ellipsis
+     * Check if $mixed is a true string value that's not empty
+     *
+     * @param $mixed
+     * @return boolean
+     */
+    public static function emptyString($mixed)
```

Comment:
Why do you have this complex function.  Isn't what you are looking for just `$string === ''`?

Reply:
TL;DR

Yes, `$string === ''` is better than `empty($string) && $string != '0'` (but that's not enough).  I'll change it to that.

Why?

I was trying to compromise to get this PR merged.  I agree it doesn't need to be this complex, though don't agree empty() is the best solution because it would return true for '0'.

preg_match() only accepts strings and when $mixed or '' is passed it produces false positives.  

In LibreNMS, I've noticed there's an abundance of empty() and preg_match() without consideration for the actual return values.  I saw a StringHelper and thought it looked like a good place to have this conversation.

1) To avoid false positives, that are breaking the bits graphs (and others breaks involving the use of preg_match() without testing the return value), we should first test if the input is a string.  

2) Then, in other cases, a second test may needed to determine if the string has no value.  Again, primarily for preg_match() but may be usable elsewhere.

Either would return true, everything else is false.  That's the emptyString() method.

$string === '' alone wouldn't meet both conditions.  It would only match true if $mixed was a string.  For example, Array() would not be considered an empty string and would be false.